feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.ts

```diff
@@ -20,32 +27,127 @@ export class ConfiguratorAttributeRadioButtonComponent
   extends ConfiguratorAttributeBaseComponent
   implements OnInit {
   attributeRadioButtonForm = new FormControl('');
+  loading$ = new BehaviorSubject<boolean>(false);
 
   @Input() attribute: Configurator.Attribute;
   @Input() ownerKey: string;
 
   @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
 
+  // TODO(#11681): make config a required dependency
+  /**
+   * default constructor
+   * @param {ConfiguratorAttributeQuantityService} quantityService
+   */
+  // eslint-disable-next-line @typescript-eslint/unified-signatures
+  constructor(quantityService: ConfiguratorAttributeQuantityService);
+
+  /**
+   * @deprecated since 3.3
+   */
+  constructor();
+
+  constructor(
+    protected quantityService?: ConfiguratorAttributeQuantityService
+  ) {
+    super();
+  }
+
   ngOnInit(): void {
     this.attributeRadioButtonForm.setValue(this.attribute.selectedSingleValue);
   }
 
-  /**
-   * Submits a value.
-   *
-   * @param {string} value - Selected value
-   */
+  get withQuantity() {
```

Comment:
this is duplicate in several places, please consider move it to some service.

Reply:
I am not sure whether I understand you correctly, @WeizhengSap. This method calls a quantityService in the method body.

Reply:
@WeizhengSap, could you have a look at this PR?

Reply:
Done!